### ISSUE TITLE
proc macro `entry` not expanded: cannot find proc-macro server

### ISSUE THREAD

**Issue by johnnybakker** [esp-rs/esp-hal#617]
I'm getting this error from rust analyzer in VSCode and NVim.

![Schermafbeelding 2023-06-23 211640](https://github.com/esp-rs/esp-hal/assets/10073000/aa5658f8-1fe4-447b-93fb-21e8333db95b)

**Comment by jessebraham** [comment]
Thanks for opening this issue!

This is actually not an issue with the HAL itself, but with how we are building and distributing the `esp` toolchain (with Xtensa support).

I have opened an issue in the `esp-rs/rust` repository requesting that the `rust-analyzer-proc-macro-srv` component is included in our distribution: https://github.com/esp-rs/rust/issues/183

I will leave this issue open in the meantime just so that others who are facing the same problem can see it, but would prefer for any additional conversation to happen in the linked issue above, please 😁

**Comment by bugadani** [comment]
Until the issue is resolved, you can see a workaround in [this repo](https://github.com/card-io-ecg/card-io-fw). In essence, I don't set the esp toolchain as default, and I have a set of xtasks to build/flash/etc using the correct toolchain.

**Comment by jessebraham** [comment]
Alternatively you can just disable the diagnostic entirely in `.vscode/settings.json`:

```json
{
  "rust-analyzer.diagnostics.disabled": ["unresolved-proc-macro"]
}
```

**Comment by jessebraham** [comment]
This issue has been resolved via [esp-rs/rust-build#215](https://github.com/esp-rs/rust-build/pull/215), so the next release of the Xtensa toolchain will contain this fix. Expect this release in the next couple weeks.

I will leave this issue open until the toolchain has been released.
